### DISCUSSION TITLE
Release/5.4.1

### DISCUSSION THREAD

**Discussion by Athanasius** [EDCD/EDMarketConnector#1597] [Announcement]
Release 5.4.1
===

* We now test against, and package with, Python 3.10.5.
* If for any reason `EDMarketConnector.exe` fails to shutdown and exit when asked to by the upgrade process this should no longer result in a spontaneous system reboot.  Closes [#1492](https://github.com/EDCD/EDMarketConnector/issues/1492).

  A manual reboot will still be required to complete the EDMarketConnector upgrade process and we make no guarantees about the stability of the application until this is done.
* The new EDDN `fsssignaldiscovered/1` schema has been implemented.
* EDSM trace level logging will no longer log API credentials unless explicitly asked to, separately from other EDSM API trace logging.

Bug Fixes
---

* EDDN: Ensure we always remove all `_Localised` suffix keys in data.  This was missed in some recent new schemas and turned out to be an issue for at least `approachsettlement/1`.

<hr /><em>This discussion was created from the release <a href='https://github.com/EDCD/EDMarketConnector/releases/tag/Release/5.4.1'>Release/5.4.1</a>.</em>

**Comment by AnvadicA** [comment]
After new update the market connector stopped to initialize the Odyssey settlements and refresh data into INARA (station update, location update)

**Reply by Athanasius** [reply]
It would be helpful if you open a [bug report](https://github.com/EDCD/EDMarketConnector/issues/new?assignees=&labels=bug%2C+unconfirmed&template=bug_report.md&title=) about this.  Be sure to include:

1. Precisely what information you're expecting to be updated on Inara, and apparently now isn't.
2. All relevant [debug log files](https://github.com/EDCD/EDMarketConnector/wiki/Troubleshooting#debug-log-files) from when you attempt this in-game.
3. The game [Journal file(s)](https://github.com/EDCD/EDMarketConnector/wiki/Troubleshooting#doesnt-auto-update-andor-persistently-gives-server-is-lagging-error) from the same time period.

**Comment by andr3w1sh** [comment]
My EDMC is no longer connecting to my EDSM, when I go to the Settings all of the options are grayed out and the Commander Name and API Key are missing.
<img width="575" alt="Screenshot 2022-08-01 132335" src="https://user-images.githubusercontent.com/32079893/182218400-91ff9403-bdf3-4200-b94d-57bce376e599.png">

**Reply by Athanasius** [reply]
> My EDMC is no longer connecting to my EDSM, when I go to the Settings all of the options are grayed out and the Commander Name and API Key are missing.

Check this *after* you've logged into the game **FULLY**.  Until then EDMC can't just assume your last used Cmdr is the one you'll next be using, so if it sees no information about that in the latest Journal file it'll be in this state.  That can be true if you last ran the game to the main menu and it's either sitting there, or you just quit out from there without going fully into the game.

Ref: https://github.com/EDCD/EDMarketConnector/wiki/Troubleshooting#credentials-settings-are-greyed-out

**Reply by andr3w1sh** [reply]
ah. that was absolutely it. I wasn't loading all the way into the game, just the starting menu. tusen takk.
